app/models/job.rb

```diff
@@ -182,13 +179,40 @@ def hash
     [self.class, id].hash
   end
 
-  def validate_array_tasks!
-    if array_range.valid?
-      array_tasks.each do |task|
-        task.validate!
-      end
-    else
-      @errors.add(:array, 'is not a valid range expression')
-    end
+  def validate_array_range
+    @errors.add(:array, 'is not a valid range expression') unless array_range.valid?
+  end
+end
+
+class Task
+  # TODO: Reinstate these validations
+  # Validations for `JOB`s and `ARRAY_JOB`s.
+  # validates :array_index,
+  #   absence: true, unless: ->() { job_type == 'ARRAY_TASK' }
+  # validates :array_job,
+  #   absence: true, unless: ->() { job_type == 'ARRAY_TASK' }
+  # Validations for `ARRAY_TASK`s
+  # validates :array_index,
+  #   presence: true,
+  #   numericality: { allow_blank: false, only_integer: true, greater_than_or_equal_to: 0 },
+  #   if: ->() { job_type == 'ARRAY_TASK' }
```

Comment:
What's needed to reinstate these?  If its quick let's do it.

Reply:
I've taken a different approach and remove the comment block entirely: https://github.com/openflighthpc/flight-scheduler-controller/pull/12/commits/226f421ff338abfc631d960fda4358557112ef1a

The `Task` objects should not be invalid as `Job` is validated up front and `TaskRegistry` is tested extensively. This means validating the `Task` gives no advantages.

Reply:
That's fine.